query/src/main/java/io/keen/client/java/DatasetDefinition.java

```diff
@@ -0,0 +1,145 @@
+package io.keen.client.java;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class DatasetDefinition {
+
+    private String datasetName;
+
+    private String displayName;
+
+    private String projectId;
+
+    private String organizationId;
+
+    private String lastScheduledDate;
+
+    private String latestSubtimeframeAvailable;
+
+    private String millisecondsBehind;
+
+    private List<String> indexBy;
+
+    private DatasetQuery query;
+
+    static DatasetDefinition fromMap(Map<String, Object> properties) {
+        DatasetDefinition definition = new DatasetDefinition();
+        definition.datasetName = (String) properties.get("dataset_name");
+        definition.displayName = (String) properties.get("display_name");
+        definition.projectId = (String) properties.get("project_id");
+        definition.organizationId = (String) properties.get("organization_id");
+        definition.lastScheduledDate = (String) properties.get("last_scheduled_date");
+        definition.latestSubtimeframeAvailable = (String) properties.get("latest_subtimeframe_available");
+        definition.millisecondsBehind = String.valueOf(properties.get("milliseconds_behind"));
+        //noinspection unchecked
+        definition.indexBy = (List<String>) properties.get("index_by");
+        //noinspection unchecked
+        definition.query = DatasetQuery.fromMap((Map<String, Object>) properties.get("query"));
+        return definition;
+    }
+
+    static List<DatasetDefinition> collectionFromMap(Map<String, Object> properties) {
```

Comment:
Why the name `collectionFromMap`? Seems very general. Also, could this be `Map<String, ?>` instead?

Reply:
I've changed it into `definitionsFromMap`